plugins/topology/src/components/Topology/TopologySideBar/Resources/IngressListSidebar.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { IngressData } from '../../../../types/ingresses';
+import TopologyResourcesTabPanelItem from '../TopologyResourcesTabPaneltem';
+import { IngressModel } from '../../../../models';
+import ResourceName from '../../../../common/components/ResourceName';
+import IngressRules from './IngressRules';
+
+const IngressListSidebar = ({
+  ingressesData,
+}: {
+  ingressesData: IngressData[];
+}) => {
+  return (
+    <TopologyResourcesTabPanelItem
+      resourceLabel={IngressModel.labelPlural}
+      dataTest="ingress-list"
+    >
+      {ingressesData?.length > 0 &&
+        ingressesData.map((ingressData: IngressData) => (
+          <li
+            className="item"
+            style={{ flexDirection: 'column' }}
+            key={ingressData.ingress.metadata?.uid}
+          >
+            <span>
+              <ResourceName
+                name={ingressData.ingress.metadata?.name ?? ''}
+                kind={ingressData.ingress.kind ?? ''}
+              />
+            </span>
+            {ingressData.url && (
+              <>
+                <span className="topology-text-muted">Location:</span>
+                <a
+                  href={ingressData.url}
+                  target="_blank"
+                  rel="noopener noreferrer"
+                >
+                  {ingressData.url}
+                </a>
+              </>
+            )}
+            {ingressData.ingress.spec?.rules?.length && (
+              <IngressRules ingress={ingressData.ingress} />
+            )}
+          </li>
+        ))}
+    </TopologyResourcesTabPanelItem>
+  );
+};
+
+export default IngressListSidebar;
```

Comment:
Can we use this comp for the routes list as well instead of having another comp, since both the comps are mostly similar except for the resource?

Reply:
have kept it separate as ingress has `IngressRules` as well which is rendered conditionally based on `ingressData.ingress.spec?.rules?.length` which is specific to ingress

Reply:
We don't have rules in Route? Also we can just have a boolean like showRules set to true in case of ingress. Wdyt?

Reply:
no route doesn’t have rules like ingress.
We can do that but that will not be much of a value add and I feel good to keep as separate unless you feel strongly otherwise.